src/librustc_mir/interpret/eval_context.rs

```diff
@@ -546,57 +566,143 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'a, 'mir, 'tcx>> InterpretCx<'a, 'mir, 'tc
         }
     }
 
-    pub(super) fn pop_stack_frame(&mut self) -> EvalResult<'tcx> {
+
+    pub(super) fn pop_stack_frame_internal(&mut self) -> EvalResult<'tcx, StackPopCleanup> {
```

Comment:
My expectation was that this function is told, via a parameter, whether this is a returning or an unwinding pop. Whoever triggers the pop should know that.

Reply:
That parameter would either be redundant (if it agrees with the tracked unwinding state), or wrong (if it disagrees with the tracked unwinding state).

Besides, I use this method in `pop_stack_frame`, when we need to pop off frames that don't contain `unwind` terminators.

Reply:
The thing that is redundant here is the tracked unwinding state. :)  This kind of indirect passing of "where do we even unwind to" makes the code very hard to follow. In particular, [this entire state machine here](https://github.com/rust-lang/rust/pull/60026/files#diff-470c35c1f39f9d3f8817435a63c73ad2R580) IMO shows that there is redundant state.

Reply:
> Besides, I use this method in pop_stack_frame, when we need to pop off frames that don't contain unwind terminators.

Frames that don't contain `Unwind` terminators should never occur during unwinding, that sounds like UB to me.